django/http/request.py

```diff
@@ -369,6 +369,20 @@ def __init__(self, environ):
                 headers[name] = value
         super().__init__(headers)
 
+    def __getitem__(self, key):
+        """
+        Django template variables cannot contain hyphens, so to allow use
```

Comment:
Strictly speaking the issue is that Django templates use what looks like attribute access to access dictionary items, so hyphenated keys are inaccessible. It is nothing to do with the variable itself.